pkg/autoops/api/api.go

```diff
@@ -879,19 +949,55 @@ func (s *AutoOpsService) isNoUpdateAutoOpsRuleCommand(req *autoopsproto.UpdateAu
 
 func (s *AutoOpsService) createUpdateAutoOpsRuleCommands(req *autoopsproto.UpdateAutoOpsRuleRequest) []command.Command {
 	commands := make([]command.Command, 0)
+
+	// The current web console uses ChangeAutoOpsRuleOpsTypeCommand to update OpsTypes and enable or disable them.
+	// So ActionType is not set and you need to update ActionType according to OpsType.
+	// ToDo: If ChangeAutoOpsRuleOpsTypeCommand is no longer used in the web console,
+	// the following code is no longer needed and should be removed.
+	var opsType *autoopsproto.OpsType = nil
 	if req.ChangeAutoOpsRuleOpsTypeCommand != nil {
 		commands = append(commands, req.ChangeAutoOpsRuleOpsTypeCommand)
+		opsType = &req.ChangeAutoOpsRuleOpsTypeCommand.OpsType
 	}
+
 	for _, c := range req.AddOpsEventRateClauseCommands {
+		if c.OpsEventRateClause.ActionType == autoopsproto.ActionType_UNKNOWN {
+			if opsType != nil && *opsType == autoopsproto.OpsType_DISABLE_FEATURE {
+				c.OpsEventRateClause.ActionType = autoopsproto.ActionType_DISABLE
+			}
```

Comment:
What if opsType is autoopsproto.OpsType_ENABLE_FEATURE? c.OpsEventRateClause.ActionType can remain unknown, so should it return an error?

Reply:
Validation is performed below.
https://github.com/bucketeer-io/bucketeer/blob/main/pkg/autoops/api/api.go#L395
The reason why we haven't validated Unknown is because the console doesn't support ActionType, so if we validate it, an error will occur, so we can't do it. Added ToDo comment.
https://github.com/bucketeer-io/bucketeer/pull/1095/commits/6fde42bc4c1879ff2e55f5002cea395130ab9778